### PR TITLE
Use numpy divide in explained variance

### PR DESCRIPTION
**Pull request by anargyri**
### Description

Useful in cases when all true labels are the same and hence explained variance should equal minus infinity.

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [x] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

File: recommenders/evaluation/spark_evaluation.py [hunk 1]
```diff
@@ -158,7 +159,8 @@ def exp_var(self):
             0
         ]
         var2 = self.y_pred_true.selectExpr("variance(label)").collect()[0][0]
-        return 1 - var1 / var2
+        # numpy divide is more tolerant to var2 being zero
+        return 1 - np.divide(var1, var2)
```
Comment (miguelgfierro) [inline recommenders/evaluation/spark_evaluation.py:163]: I'm looking at the test here: https://github.com/microsoft/recommenders/blob/main/tests/unit/recommenders/evaluation/test_spark_evaluation.py#L239, should we maybe add the edge case for when var2 is zero?

**Review by miguelgfierro — APPROVED** [review]

File: recommenders/evaluation/spark_evaluation.py (hunk 1 above)
Comment (anargyri) [reply recommenders/evaluation/spark_evaluation.py:163]: Maybe, although it is a very unusual case in practice (everything in the data is rated equally).
